cupy/lib/_routines_poly.py

```diff
@@ -71,3 +71,43 @@ def polysub(a1, a2):
         out = out.astype(cupy.result_type(a1, a2), copy=False)
         out -= 2 * out - a1
     return out
+
+
+def roots(p):
+    """Computes the roots of a polynomial with given coefficients.
+
+    Args:
+        p (cupy.ndarray): polynomial coefficients.
+
+    Returns:
+        cupy.ndarray: polynomial roots.
+
+    .. warning::
+
+        This function doesn't support currently polynomial coefficients
+        whose companion matrices are general 2d square arrays. Only those
+        with complex Hermitian or real symmetric 2d arrays are allowed.
+
+    .. seealso:: :func:`numpy.roots`
+
+    """
+    if p.ndim == 0:
+        raise TypeError('0-dimensional input is not allowed')
+    if p.size < 2 or (p.ndim == 1 and not cupy.any(p)):
+        return cupy.array([])
+    [p] = cupy.polynomial.polyutils.as_series([p[::-1]])
+    if p.size < 2:
+        return cupy.array([])
+    if p.size == 2:
+        out = cupy.array([-p[0]/p[1]])
+        if p[0] == 0:
+            out = out.real.astype(float)
```

Comment:
For the readability.
```suggestion
            out = out.real.astype(numpy.float64)
```